Since commit 7eb8eecbb (ash: eval: Add assignment built-in support
again) building BusyBox with the 'command' built-in disabled fails.

parse_command_args() only needs to be called when the 'command'
built-in is run.  Which it won't be if it's disabled.

v2: Avoiding infinite loops is good, too.  Thanks, Harald van Dijk.

Reported-by: Deweloper <deweloper at wp.pl>
Signed-off-by: Ron Yorston <rmy at pobox.com>
---
 shell/ash.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/shell/ash.c b/shell/ash.c
index d0d99f60e..a11b1d67d 100644
--- a/shell/ash.c
+++ b/shell/ash.c
@@ -8675,7 +8675,6 @@ typecmd(int argc UNUSED_PARAM, char **argv)
 	return err;
 }
 
-#if ENABLE_ASH_CMDCMD
 static struct strlist *
 fill_arglist(struct arglist *arglist, union node **argpp)
 {
@@ -8692,6 +8691,7 @@ fill_arglist(struct arglist *arglist, union node **argpp)
 	return *lastp;
 }
 
+#if ENABLE_ASH_CMDCMD
 /* Is it "command [-p] PROG ARGS" bltin, no other opts? Return ptr to "PROG" if yes */
 static int
 parse_command_args(struct arglist *arglist, union node **argpp, const char **path)
@@ -10190,11 +10190,13 @@ evalcommand(union node *cmd, int flags)
 				vlocal = !spclbltin;
 			}
 			cmd_is_exec = cmdentry.u.cmd == EXECCMD;
+#if ENABLE_ASH_CMDCMD
 			if (cmdentry.u.cmd != COMMANDCMD)
 				break;
 
 			cmd_flag = parse_command_args(&arglist, &argp, &path);
 			if (!cmd_flag)
+#endif
 				break;
 		}
 
-- 
2.24.1
